### PR TITLE
Update react-scripts package

### PR DESCRIPTION
**Pull request by minevala**
## Description
- Update react-scripts package
- Update react-scripts package's dependencies:
   - Update @storybook/preset-create-react-app
      - Use webpack 5 to fix new preset build
- Update eslint packages to fix linting after package updates
  - Fix some errors and omit unnecessary rules
   
## Related Issue
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-1146

## How Has This Been Tested?
- Locally on dev machine

### CODE REVIEW

File: packages/react/src/components/cookieConsent/contexts/ContentContext.tsx [hunk 1]
```diff
@@ -73,11 +73,13 @@ export type Callbacks = {
   onConsentsParsed?: (consents: ConsentObject, hasUserHandledAllConsents: boolean) => void;
 };
 
+/* eslint-disable */
 type ConsentContextProps = {
   cookieDomain?: string;
   children: React.ReactNode | React.ReactNode[] | null;
   contentSource: CookieContentSource;
 };
+/* eslint-enable */
```
Comment (harriplappalainen) [inline packages/react/src/components/cookieConsent/contexts/ContentContext.tsx:82]: Does this part mess up eslint for some reason? 😆

**Review by harriplappalainen — APPROVED** [review]
You made it work it! In record time! A medal for you, good sir 🏅 

I tested this locally and built and ran core, react and site and everything seemed to work without any errors. There are some quirky fixes to warnings and eslint errors but I think we can live with those minor beauty errors since this works.

File: packages/react/src/components/cookieConsent/contexts/ContentContext.tsx (hunk 1 above)
Comment (minevala) [reply packages/react/src/components/cookieConsent/contexts/ContentContext.tsx:82]: It should not. It just enables it again after a few lines of being disabled

Comment (harriplappalainen) [reply packages/react/src/components/cookieConsent/contexts/ContentContext.tsx:82]: Yes I meant the disabled part that it's funny and weird that apparently it enrages eslint. Nothing wrong with this, I was just curious :)

File: packages/react/.eslintrc.json [hunk 2]
```diff
@@ -43,6 +43,12 @@
     "react/jsx-one-expression-per-line": "off",
     "react/jsx-props-no-spreading": "off",
     "react/jsx-curly-newline": "off",
+    "react/default-props-match-prop-types": [0],
+    "react/destructuring-assignment": [0],
+    "react/function-component-definition": [0],
+    "react/jsx-no-constructed-context-values": [0],
+    "react/no-unstable-nested-components": [0],
+    "react/require-default-props": [0],
```
Comment (minevala) [inline packages/react/.eslintrc.json:51]: // eslint-plugin-react updated destructuring-assignment
    "react/destructuring-assignment": [0],
    // eslint-plugin-react updated function-component-definition
    "react/function-component-definition": [0],
    // new in eslint-plugin-react
    "react/jsx-no-constructed-context-values": [0],
    // new in eslint-plugin-react
    "react/no-unstable-nested-components": [0],
    // eslint-plugin-react requires now props
    "react/require-default-props": [0],
    "react/default-props-match-prop-types": [0],

**Pull request merged**